File: Easy/Java/771. Jewels and Stones.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        // Create a set to contain the different types of jewels
        // Count of jewels amongst the stones
        Set<Character> set = new HashSet<Character> ();
        int count = 0;
        
        // Add the different jewels into the set
        for(int i = 0; i < jewels.length(); i++){
            set.add(jewels.charAt(i));
        }

        // Go over the stones, looking for jewels
        for(int i = 0; i < stones.length(); i++){
            if(set.contains(stones.charAt(i)))
                count++;
        }
        
        // Return the count of jewels
        return count;
    }
}

File: Easy/Python/771. Jewels and Stones.py
class Solution(object):
    def numJewelsInStones(self, jewels, stones):
        # Create a dictionary to contain the different types of jewels
        # Count of jewels amongst the stones
        types = {}
        count = 0
        
        # Add the different jewels into the dictionary
        for i in range(len(jewels)):
            types[i] = jewels[i]
        
        # Go over the stones, looking for jewels
        for i in range(len(stones)):
            if stones[i] in types.values():
                count += 1
        
        # Return the count of jewels
        return count
